main.py

```diff
@@ -1,122 +1,190 @@
-import os
-import paramiko
+"""This module is used to check GPU machines information."""
+
+from typing import NamedTuple
 import xml.etree.ElementTree as ET
+import os
 import re
 import time
-from reprint import output
+import paramiko
+import reprint
 
-JUMP_SHELL = 'shell4.doc.ic.ac.uk'
+# Change these according to your own configuration.
 USERNAME = 'afs219'
-SSH_KEY_LOC = os.getenv('HOME') + '/.ssh/id_rsa'
-SERVERS = ['gpu0{}'.format(x) for x in range(1, 10)] + ['gpu{}'.format(x) for x in range(10,30)] +  ['RAY0{}'.format(x) for x in range(1, 10)] + ['RAY{}'.format(x) for x in range(10,27)]
+SSH_KEY_PATH = os.path.expanduser('~') + '/.ssh/id_rsa'
+
+JUMP_SHELL = 'shell4.doc.ic.ac.uk'  # One of shell1, shell2, shell3, shell4 or shell5
 SSH_TIMEOUT = 1
 REFRESH_RATE = 5
 
-def main():
-    gpu_checker = GPUChecker(SERVERS)
-    gpu_checker.run()
+# Lab Machines
+TEXELS  = [f'texel{x:02}'  for x in range(1, 45)]
+SPRITES = [f'sprite{x:02}' for x in range(1, 39)]
+ARCS    = [f'arc{x:02}'    for x in range(1, 15)]
+GPUS    = [f'gpu{x:02}'    for x in range(1, 28)]
+EDGES   = [f'edge{x:02}'   for x in range(1, 41)]
+VERTEXS = [f'vertex{x:02}' for x in range(1, 63)]
+RAYS    = [f'ray{x:02}'    for x in range(1, 27)]
+POINTS  = [f'point{x:02}'  for x in range(1, 30)]
+
+SERVERS = GPUS
+
+
+class GPUInfo(NamedTuple):
```

Comment:
This is a nice construction. I have partially written a revised version which uses the rich library for a much better display of GPU information, and this will definitely be good to have their as the __repr__ can be a row!